src/go/rdctl/cmd/shell.go

```diff
@@ -99,3 +110,57 @@ func setupLimaHome() error {
 	os.Setenv("LIMA_HOME", candidatePath)
 	return nil
 }
+
+const restartDirective = "Either run 'rdctl start' or start the Rancher Desktop application first"
+
+func checkLimaIsRunning(commandName string) bool {
+	output, err := exec.Command(commandName, "ls", "0", "--format", "{{.Status}}").CombinedOutput()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to run 'rdctl shell': %s\n", err)
+		return false
+	}
+	if strings.HasPrefix(string(output), "Running") {
+		return true
+	}
+
+	if output == nil || strings.Contains(string(output), "No instance matching 0 found") {
+		fmt.Fprintf(os.Stderr, "The Rancher Desktop VM needs to be created.\n%s.\n", restartDirective)
+	} else {
+		fmt.Fprintf(os.Stderr,
+			"The Rancher Desktop VM needs to be in state \"Running\" in order to execute 'rdctl shell', but it is currently in state \"%s\".\n%s.\n", strings.TrimRight(string(output), "\n"), restartDirective)
+	}
+	return false
+}
+
+func checkWSLIsRunning(distroName string) bool {
+	cmd := exec.Command("wsl", "-l", "-v")
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to run 'wsl -l': %s\n", err)
+		return false
+	}
+	isListed := false
+	targetState := ""
+	for _, line := range regexp.MustCompile(`\r?\n`).Split(string(output), -1) {
+		fields := regexp.MustCompile(`\s+`).Split(line, -1)
+		if fields[0] == "*" {
+			fields = fields[1:]
+		}
+		if len(fields) >= 2 && fields[0] == distroName {
+			isListed = true
+			targetState = fields[1]
+			break
+		}
+	}
+	if targetState == "Running" {
```

Comment:
Note to self: `wsl --list --verbose` doesn't appear to actually translate the status names; I've checked fr-CA and zh-TW, both use `Running` / `Stopped`.  So this is probably okay.

(I've found that trying to access `\\wsl$\<distro>` to be a nice way to check things without depending on text; but potentially MS could fix that in the future too…)

Reply:
I'll leave that code as is.